Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -310,6 +328,12 @@ public sealed class SandboxedProcessPipExecutor : ISandboxedProcessFileStorage
             }
 
             m_vmInitializer = vmInitializer;
+
+            m_incrementalToolFragments = new List<string>(

```

Comment:
So cryptic with that `??` operator hidden in the back.  :)

This would look less cryptic to me:
```csharp
m_incrementalToolFragments = (incrementalTools ?? Enumerable.Empty<string>())
  .Select(...)
  .ToList();
```

Reply:
Agreed; however, yours give some compilation errors. To fix them, it will get things complicated. i just added if/else branch.

---
In reply to: [310279032](https://github.com/microsoft/BuildXL/pull/697#discussion_r310279032) [](ancestors = 310279032)